localstack/services/apigateway/helpers.py

```diff
@@ -130,8 +131,14 @@ class OpenAPIExt:
 # TODO: make the CRUD operations in this file generic for the different model types (authorizes, validators, ...)
 
 
-def get_apigateway_store(account_id: str = None, region: str = None) -> ApiGatewayStore:
-    return apigateway_stores[account_id or get_aws_account_id()][region or aws_stack.get_region()]
+def get_apigateway_store(
```

Comment:
another option would be to have two method overloading one with the existing interface and another with the context. That because if the only thing we care is about those two fields it's a bit redundant and we can eventually make the request context (RequestContext) present in the `ApigatewayContext` class so can be used everywhere and not only on the provider where it originates, and migrate towards that signature `get_apigateway_store(context: RequestContext)`

Reply:
Good point, agreed. Updated the PR and unified the store access a bit further. Turns out we only need two variants of get_store - one for `RequestContext`, one for `ApiInvocationContext`. I think this makes it more clear and allows us to converge towards cleaner signatures - thanks for the pointer! 👍